### PR TITLE
Rashad walldoors

### PR DESCRIPTION
**Pull request by rashadg1030**
Changed doors to Map (Int, Int) Int and walls to a set of (Int, Int). Changed some functions to accommodate changes.

### CODE REVIEW

**Comment by rashadg1030** [comment]
Got wall collision to work but there's probably a better way to handle it.

File: src/Day1.hs [hunk 1]
```diff
@@ -105,9 +110,11 @@ update input st
   -- FOR DEUGGING ABOVE
   | otherwise = case scene st of
       Scene'Start -> return $ updateStart input st
+      -- Scene'Play -> return $ st
```
Comment (jxv) [inline src/Day1.hs:113]: Just delete the code

File: src/Day1.hs [hunk 2]
```diff
@@ -127,9 +134,22 @@ updateGameOver input st = st
   where
     isStart = lookupKey (keys input) Enter == Pressed || lookupKey (keys input) (Char ' ') == Pressed
 
-updatePlayer :: Input -> Player -> Player
-updatePlayer input p = p
-  { playerPos = updatePlayerPos input (playerPos p) }
+updatePlayer :: Input -> PlayState -> PlayState
```
Comment (jxv) [inline src/Day1.hs:137]: rename to `updatePlayState`

File: src/Day1.hs [hunk 3]
```diff
@@ -127,9 +134,22 @@ updateGameOver input st = st
...
+  where
+    p = player ps
+    currPos = playerPos p
+    nextPos = updatePlayerPos input currPos  
+    roomIndex = playerRoom p
+    room = fromJust $ lookupMap roomIndex (rooms ps)
+    wPos = walls room
+    dPos = undefined  
+
+-- updatePlayer :: Input -> Player -> Player
```
Comment (jxv) [inline src/Day1.hs:150]: delete

File: src/Day1.hs [hunk 4]
```diff
@@ -127,9 +134,22 @@ updateGameOver input st = st
...
+  True  -> ps
+  False -> ps { player = p { playerPos = updatePlayerPos input (playerPos p) } }
+  where
+    p = player ps
+    currPos = playerPos p
+    nextPos = updatePlayerPos input currPos  
+    roomIndex = playerRoom p
+    room = fromJust $ lookupMap roomIndex (rooms ps)
+    wPos = walls room
+    dPos = undefined
```
Comment (jxv) [inline src/Day1.hs:148]: delete `dPos` if it's not being used and it's `undefined`

File: src/Day1.hs [hunk 5]
```diff
@@ -23,24 +25,26 @@ data Scene
 
 data PlayState = PlayState
   { player :: Player
-  , rooms :: Map RoomIndex Room
+  , rooms :: M.Map RoomIndex Room
```
Comment (jxv) [inline src/Day1.hs:28]: `M.` is unneed. It's already in scope from grid-proto's rexport.

**Pull request merged**